app/src/main/java/org/hackillinois/android/view/ScannerFragment.kt

```diff
@@ -1,215 +1,145 @@
-    }
-
-    /**
-     * Make the switch invisible when not relevant to the event.
-     */
-    private fun updateOverrideSwitchVisibility(it: Boolean?) {
-        staffOverrideSwitch.visibility = when (it) {
-            true -> AdapterView.VISIBLE
-            false -> AdapterView.GONE
-            null -> AdapterView.GONE
+        viewModel = ViewModelProviders.of(this).get(ScannerViewModel::class.java).apply {
+            init()
+            eventsListLiveData.observe(this@ScannerFragment, Observer { updateEventList(it) })
+            lastScanStatus.observe(this@ScannerFragment, Observer { displayScanResult(it) })
+            shouldDisplayOverrideSwitch.observe(this@ScannerFragment, Observer { updateOverrideSwitchVisibility(it) })
         }
     }
 
-    /**
-     * Called when scanner scans something.
-     */
-    private var onQrCodeScan: BarcodeCallback = object : BarcodeCallback {
-        override fun possibleResultPoints(resultPoints: MutableList<ResultPoint>?) {
-        }
-
-        override fun barcodeResult(result: BarcodeResult) {
-            if (shouldParseScan) {
-                // stop parsing scans until we have finished this one!
-                shouldParseScan = false
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.fragment_scanner, container, false)
 
-                val currentEvent = eventListView.selectedItem.toString()
+        view.eventListView.onItemSelectedListener = viewModel.onEventSelected
 
-                // Prevent duplicate scans
-                if (getUserIdFromQrString(result.text) == lastSuccessfullyScannedUser && currentEvent == lastEventScannedFor) {
-                    return
+        context?.let { context ->
+            codeScanner = CodeScanner(context, view.codeScannerView).apply {
+                camera = CodeScanner.CAMERA_BACK
+                formats = listOf(BarcodeFormat.QR_CODE)
+                autoFocusMode = AutoFocusMode.SAFE
+                scanMode = ScanMode.SINGLE
+                isAutoFocusEnabled = true
+                isFlashEnabled = false
+                decodeCallback = DecodeCallback {
+                    val event = view.eventListView.selectedItem.toString()
+                    val userId: String = getUserIdFromQrString(it.text)
+                    viewModel.checkUserIntoEvent(event, userId, view.staffOverrideSwitch.isChecked)
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(context, "Sending request to API. Please wait for a response", Toast.LENGTH_LONG).show()
+                    }
                 }
+                errorCallback = ErrorCallback {
+                    Handler(Looper.getMainLooper()).post {
+                        Toast.makeText(context, "Camera initialization error: ${it.message}", Toast.LENGTH_LONG).show()
```

Comment:
How verbose could these errors be? Will they even fit inside of a toast?

Reply:
From the source it looks like "Unable to access camera" and "Unable to configure camera" are the only two possibilities. I'll remove the beginning string since it is redundant, and that should be fine.